cpp/src/arrow/python/python_to_arrow.cc

```diff
@@ -246,6 +254,7 @@ class SequenceBuilder {
   std::vector<int32_t> list_offsets_;
   std::vector<int32_t> tuple_offsets_;
   std::vector<int32_t> dict_offsets_;
+  std::vector<int32_t> set_offsets_;
```

Comment:
Would there be any benefits to commingling the sequence data in a single union versus splitting up lists, tuples, and sets?